presto-main/src/main/java/io/prestosql/sql/planner/PlanOptimizers.java

```diff
@@ -307,19 +307,30 @@ public PlanOptimizers(
                                         new PushLimitThroughMarkDistinct(),
                                         new PushLimitThroughOuterJoin(),
                                         new PushLimitThroughSemiJoin(),
-                                        new PushLimitIntoTableScan(metadata),
-                                        new PushPredicateIntoTableScan(metadata, typeAnalyzer),
-                                        new PushSampleIntoTableScan(metadata),
                                         new RemoveTrivialFilters(),
                                         new ImplementFilteredAggregations(),
                                         new SingleDistinctAggregationToGroupBy(),
                                         new MultipleDistinctAggregationToMarkDistinct(),
-                                        new ImplementBernoulliSampleAsFilter(),
                                         new MergeLimitWithDistinct(),
                                         new PruneCountAggregationOverScalar(),
                                         new PruneOrderByInAggregation(metadata.getFunctionRegistry()),
                                         new RewriteSpatialPartitioningAggregation(metadata)))
                                 .build()),
+                new IterativeOptimizer(
+                        ruleStats,
+                        statsCalculator,
+                        estimatedExchangesCostCalculator,
+                        ImmutableSet.of(
+                                new PushLimitIntoTableScan(metadata),
```

Comment:
Don't move these. There are benefits to having them in the same loop as the previous rules.